### PR TITLE
added the new ns-convert

### PR DESCRIPTION
**Pull request by abhay447**
The new ns-convert uses the nix format instead of hdf5 via nixpy. This one is a basic version and is able to convert the sample file provided with mcd library.

### CODE REVIEW

File: ns-convert [hunk 1]
```diff
@@ -1,184 +1,341 @@
...
-    sys.exit(res)
+#!/usr/bin/env python
+from __future__ import print_function
+
+import os
+import sys
+import neuroshare as ns
+import numpy as np
+import time
+import nix
```
Comment (gicmo) [inline ns-convert]: In newer versions of nixpy the package is called `nixio`.

**Comment by gicmo** [comment]
In general: `ns-convert` somehow lost its executable bit. Also the encoding is now dos(!). It really should be utf-8[unix]. A few trailing whitespace are also present and should be removed.

File: ns-convert [hunk 2]
```diff
@@ -1,184 +1,341 @@
...
+        sys.stdout.write(msg)
+        sys.stdout.flush()
+
+    def cleanup(self):
+        sys.stdout.write('%s\r' % (' '*len(self._last_msg)))
+        sys.stdout.flush()
+
+
+def main():
+    global Converter,H5Converter
```
Comment (gicmo) [inline ns-convert]: That should not be necessary.

File: ns-convert [hunk 3]
```diff
@@ -1,184 +1,341 @@
...
+    output = None
+    parser = argparse.ArgumentParser(description='Convert Neuroshare readable files to nix/hdf5 format')
+    parser.add_argument("-o","--output",type=str,help="name of the output file")
+    parser.add_argument("--hdf5",action="store_true",help="For plain conversion to hdf5 format")
+    parser.add_argument("InputFile",type=str,help="File to be converted")
+    args = parser.parse_args()
+    
+    if args.output:
```
Comment (gicmo) [inline ns-convert]: Not necessary, `argparse` will enforce  that you have a `InputFile` argument.

File: ns-convert [hunk 4]
```diff
@@ -1,184 +1,341 @@
...
+    if args.hdf5:
+        Converter = H5Converter
+    
+    filename = args.InputFile
+    ci = ConsoleIndicator()
+    converter = Converter(filename, output, progress=ci)
+    start = time.time()
+    converter.convert()
+    ci.cleanup()
+    print("converted in "+str(time.time()-start)+" seconds")
```
Comment (gicmo) [inline ns-convert]: Way to many digits :) "converted in 2.24568319321 seconds" maybe round to X.Y seconds? Also print it to standard error not standard out.

File: ns-convert [hunk 5]
```diff
@@ -1,184 +1,341 @@
...
+    parser = argparse.ArgumentParser(description='Convert Neuroshare readable files to nix/hdf5 format')
+    parser.add_argument("-o","--output",type=str,help="name of the output file")
+    parser.add_argument("--hdf5",action="store_true",help="For plain conversion to hdf5 format")
+    parser.add_argument("InputFile",type=str,help="File to be converted")
+    args = parser.parse_args()
+    
+    if args.output:
+        output = args.output
+    if args.hdf5:
+        Converter = H5Converter
```
Comment (gicmo) [inline ns-convert]: Don't overwrite converter class.
Just do a `converter = H5Converter if args.hdf5 else Converter` in like 331.

File: ns-convert [hunk 6]
```diff
@@ -1,184 +1,341 @@
...
+    def __add__(self, other):
+        self._cur_value += other
+        self.progress(self._max_value, self._cur_value)
+        return self
+
+    def progress(self, max_value, cur_value):
+        pass
+
+#CLASS DEFINITION FOR NIX(Default) CONVERSION
+class Converter(object):
```
Comment (gicmo) [inline ns-convert]: Call this `NIXConverter`?

File: ns-convert [hunk 7]
```diff
@@ -1,184 +1,341 @@
...
+        Converter = H5Converter
+    
+    filename = args.InputFile
+    ci = ConsoleIndicator()
+    converter = Converter(filename, output, progress=ci)
+    start = time.time()
+    converter.convert()
+    ci.cleanup()
+    print("converted in "+str(time.time()-start)+" seconds")
+    return 0
```
Comment (gicmo) [inline ns-convert]: Maybe also print the output filename to stderr?

File: ns-convert [hunk 8]
```diff
@@ -1,184 +1,341 @@
...
+#CLASS DEFINITION FOR NIX(Default) CONVERSION
+class Converter(object):
+    def __init__(self, filepath, output=None, progress=None):
+        if not output:
+            (basefile, ext) = os.path.splitext(filepath)
+            output = "%s.h5" % basefile
+
+        nf = ns.File(filepath)
+        nixF = nix.File.open(output,nix.FileMode.Overwrite)
+        nixB = nixF.create_block("main_block","nix.session")
```
Comment (gicmo) [inline ns-convert]: `main_block` and `main_section` are not nice names. Not sure what else to use. @jgrewe any ideas? maybe the filename for both?

File: ns-convert [hunk 9]
```diff
@@ -1,184 +1,341 @@
...
+
+#CLASS DEFINITION FOR NIX(Default) CONVERSION
+class Converter(object):
+    def __init__(self, filepath, output=None, progress=None):
+        if not output:
+            (basefile, ext) = os.path.splitext(filepath)
+            output = "%s.h5" % basefile
+
+        nf = ns.File(filepath)
+        nixF = nix.File.open(output,nix.FileMode.Overwrite)
```
Comment (gicmo) [inline ns-convert]: space between ",nix.File..." :)

File: ns-convert [hunk 10]
```diff
@@ -1,184 +1,341 @@
...
+        
+    def get_section_for_type(self, entity_type):
+        name_map = {1: 'events',
+                    2: 'analogs',
+                    3: 'segments',
+                    4: 'neurals'}
+
+        if entity_type not in self._sections:
+            name    = name_map[entity_type]
+            section = self._nixS.create_section(name,"odml.recording")
```
Comment (gicmo) [inline ns-convert]: I like the `=` aligned too, but that will give a lot of PEP-8 warnings in a modern IDE, so let's rather not align them.

File: ns-convert (hunk 8 above)
Comment (jgrewe) [inline ns-convert]: is it correct that ns files always represent a single dataset? single recording? If true then the name of the dataset would be fine, I guess.

Comment (gicmo) [inline ns-convert]: @jgrewe I think so, yes. But I am not sure there is a strong requirement written down somewhere. :)

Comment (jgrewe) [inline ns-convert]: ok, cool. :) But then the main_block, etc. would also fail.

File: ns-convert (hunk 4 above)
Comment (jgrewe) [inline ns-convert]:
````
"converted in %.5f seconds" % time.time() - start
````

**Comment by gicmo** [comment]
I am afraid you have converted the file back to dos encoding ;( Could you quickly fix that? ;)

**Comment by gicmo** [comment]
There are still some minor issues left, but we can fix them on master. Merged! ;)

**Pull request merged**